test/test_edit.py

```diff
@@ -0,0 +1,172 @@
+from beetsplug.edit import EditPlugin
+
+
+class ModifyFileMocker(object):
+    """Helper for modifying a file, replacing or editing its contents. Used for
+    mocking the calls to the external editor during testing."""
+
+    def __init__(self, contents=None, replacements=None):
+        """ `self.contents` and `self.replacements` are initalized here, in
+        order to keep the rest of the functions of this class with the same
+        signature as `EditPlugin.get_editor()`, making mocking easier.
+            - `contents`: string with the contents of the file to be used for
+            `overwrite_contents()`
+            - `replacement`: dict with the in-place replacements to be used for
+            `replace_contents()`, in the form {'previous string': 'new string'}
+
+        TODO: check if it can be solved more elegantly with a decorator
+        """
+        self.contents = contents
+        self.replacements = replacements
+        self.action = self.overwrite_contents
+        if replacements:
+            self.action = self.replace_contents
+
+    def overwrite_contents(self, filename):
+        """Modify `filename`, replacing its contents with `self.contents`. If
+        `self.contents` is empty, the file remains unchanged.
+        """
+        if self.contents:
+            with codecs.open(filename, 'w', encoding='utf8') as f:
+                f.write(self.contents)
+
+    def replace_contents(self, filename):
+        """Modify `filename`, reading its contents and replacing the strings
+        specified in `self.replacements`.
+        """
+        with codecs.open(filename, 'r', encoding='utf8') as f:
+            contents = f.read()
+        for old, new_ in self.replacements.iteritems():
+            contents = contents.replace(old, new_)
+        with codecs.open(filename, 'w', encoding='utf8') as f:
+            f.write(contents)
+
+
+class EditCommandTest(unittest.TestCase, TestHelper):
+    """ Black box tests for `beetsplug.edit`. Command line interaction is
+    simulated using `test.helper.control_stdin()`, and yaml editing via an
+    external editor is simulated using `ModifyFileMocker`.
+    """
+    ALBUM_COUNT = 1
+    TRACK_COUNT = 10
+
+    def setUp(self):
+        self.setup_beets()
+        self.load_plugins('edit')
+        # make sure that we avoid invoking the editor except for making changes
+        self.config['edit']['diff_method'] = ''
+        # add an album
+        self.add_album_fixture(track_count=self.TRACK_COUNT)
+        # keep track of write()s
+        library.Item.write = Mock()
+
+    def tearDown(self):
+        self.teardown_beets()
+        self.unload_plugins()
+
+    def run_mocked_command(self, modify_file_args={}, stdin=[]):
+        """Run the edit command, with mocked stdin and yaml writing."""
+        m = ModifyFileMocker(**modify_file_args)
+        with patch.object(EditPlugin, 'get_editor', side_effect=m.action):
+            with control_stdin('\n'.join(stdin)):
```

Comment:
Yes, I think it's the canonical way to do it. After all, since stdin is just a stream, new lines are what define separate prompt responses.

I suppose one could also mock raw_input as an alternative.